<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Enigma</class>
 <widget class="QMainWindow" name="Enigma">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Enigma</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QGroupBox" name="config_groupBox">
      <property name="title">
       <string>Config</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="config_verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="configHeader_horizontalLayout">
           <item>
            <widget class="QCheckBox" name="typex_checkBox">
             <property name="text">
              <string>Typex</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_8">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="reload_pushButton">
             <property name="text">
              <string>Reload from File</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="rotor_widget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QGridLayout" name="rotors_gridLayout">
              <item row="0" column="4">
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="3">
               <widget class="QComboBox" name="rotor2_comboBox"/>
              </item>
              <item row="2" column="3">
               <widget class="QSpinBox" name="rotor2_spinBox"/>
              </item>
              <item row="2" column="9">
               <widget class="QSpinBox" name="rotor5_spinBox"/>
              </item>
              <item row="0" column="5">
               <widget class="QLabel" name="rotor3_label">
                <property name="text">
                 <string>Rotor 3</string>
                </property>
               </widget>
              </item>
              <item row="0" column="8">
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="0" column="6">
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="0" column="9">
               <widget class="QLabel" name="rotor5_label">
                <property name="text">
                 <string>Rotor 5</string>
                </property>
               </widget>
              </item>
              <item row="0" column="10">
               <spacer name="horizontalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="7">
               <widget class="QComboBox" name="rotor4_comboBox"/>
              </item>
              <item row="1" column="1">
               <widget class="QComboBox" name="rotor1_comboBox"/>
              </item>
              <item row="0" column="0">
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="9">
               <widget class="QComboBox" name="rotor5_comboBox"/>
              </item>
              <item row="0" column="7">
               <widget class="QLabel" name="rotor4_label">
                <property name="text">
                 <string>Rotor 4</string>
                </property>
               </widget>
              </item>
              <item row="2" column="5">
               <widget class="QSpinBox" name="rotor3_spinBox"/>
              </item>
              <item row="0" column="2">
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="rotor1_label">
                <property name="text">
                 <string>Rotor 1</string>
                </property>
               </widget>
              </item>
              <item row="2" column="7">
               <widget class="QSpinBox" name="rotor4_spinBox"/>
              </item>
              <item row="1" column="5">
               <widget class="QComboBox" name="rotor3_comboBox"/>
              </item>
              <item row="0" column="3">
               <widget class="QLabel" name="rotor2_label">
                <property name="text">
                 <string>Rotor 2</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="rotor1_spinBox"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="plugboard_groupBox">
           <property name="title">
            <string>Plugboard</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QScrollArea" name="plugboard_scrollArea">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>743</width>
                 <height>238</height>
                </rect>
               </property>
               <layout class="QFormLayout" name="formLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="wire1_label">
                  <property name="text">
                   <string>Wire 1</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLineEdit" name="wire1_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="wire2_label">
                  <property name="text">
                   <string>Wire 2</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="wire2_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="wire3_label">
                  <property name="text">
                   <string>Wire 3</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLineEdit" name="wire3_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="wire4_label">
                  <property name="text">
                   <string>Wire 4</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QLineEdit" name="wire4_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="wire5_label">
                  <property name="text">
                   <string>Wire 5</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QLineEdit" name="wire5_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="5" column="0">
                 <widget class="QLabel" name="wire6_label">
                  <property name="text">
                   <string>Wire 6</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QLineEdit" name="wire6_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="6" column="0">
                 <widget class="QLabel" name="wire7_label">
                  <property name="text">
                   <string>Wire 7</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="1">
                 <widget class="QLineEdit" name="wire7_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="7" column="0">
                 <widget class="QLabel" name="wire8_label">
                  <property name="text">
                   <string>Wire 8</string>
                  </property>
                 </widget>
                </item>
                <item row="7" column="1">
                 <widget class="QLineEdit" name="wire8_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="8" column="0">
                 <widget class="QLabel" name="wire9_label">
                  <property name="text">
                   <string>Wire 9</string>
                  </property>
                 </widget>
                </item>
                <item row="8" column="1">
                 <widget class="QLineEdit" name="wire9_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
                <item row="9" column="0">
                 <widget class="QLabel" name="wire10_label">
                  <property name="text">
                   <string>Wire 10</string>
                  </property>
                 </widget>
                </item>
                <item row="9" column="1">
                 <widget class="QLineEdit" name="wire10_lineEdit">
                  <property name="maxLength">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="input_groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Input</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QPlainTextEdit" name="input_plainTextEdit"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="output_groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Output</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPlainTextEdit" name="output_plainTextEdit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="encode_pushButton">
           <property name="text">
            <string>Encode</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="copy_pushButton">
           <property name="text">
            <string>Copy</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionEncode_Input"/>
    <addaction name="separator"/>
    <addaction name="actionOpen_Config"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionEnigma_Help"/>
    <addaction name="separator"/>
    <addaction name="actionAbout_Enigma"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionPaste_Message"/>
    <addaction name="actionCopy_Output"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen_Config">
   <property name="text">
    <string>Open Config</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
  </action>
  <action name="actionPaste_Message">
   <property name="text">
    <string>Paste Message</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+V</string>
   </property>
  </action>
  <action name="actionCopy_Output">
   <property name="text">
    <string>Copy Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionEnigma_Help">
   <property name="text">
    <string>Enigma Help</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+/</string>
   </property>
  </action>
  <action name="actionAbout_Enigma">
   <property name="text">
    <string>About Enigma</string>
   </property>
  </action>
  <action name="actionEncode_Input">
   <property name="text">
    <string>Encode Input</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
